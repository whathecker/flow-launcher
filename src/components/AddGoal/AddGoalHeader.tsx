import React from "react";
import { Image, StyleSheet } from "react-native";
import { View, Text } from "../Themed";
import { CloseIcon } from "../shared";

import { Container, Typography } from "../../styles";

type AddGoalHeaderProps = {
  backBtnHandler: () => void;
};

const AddGoalHeader: React.FC<AddGoalHeaderProps> = ({
  backBtnHandler,
}: AddGoalHeaderProps) => {
  return (
    <>
      <View style={styles.closeIconWrapper}>
        <CloseIcon pressHandler={backBtnHandler} />
      </View>
      <View style={styles.headerWrapper}>
        <Text
          style={styles.headerText}
          lightColor="#554F4F"
          darkColor="#554F4F"
        >
          {`New Goal`}
        </Text>
        <Image
          style={styles.rocketImage}
          source={require(`../../../assets/images/rocket_1f680.png`)}
        />
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  closeIconWrapper: {
    paddingLeft: 25,
    paddingTop: 0,
  },
  headerWrapper: {
    ...Container.flexStart,
    paddingLeft: 15,
    paddingTop: 80,
  },
  rocketImage: {
    width: 32,
    height: 32,
  },
  headerText: {
    ...Typography.h1,
    paddingRight: 12,
  },
});

export default AddGoalHeader;
