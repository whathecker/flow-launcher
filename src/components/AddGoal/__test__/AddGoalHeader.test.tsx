import * as React from "react";
import renderer from "react-test-renderer";
import AddGoalHeader from "../AddGoalHeader";

describe("Test AddGoalHeader component", () => {
  it("Should renders AddGoalHeader correctly", () => {
    const tree = renderer
      .create(
        <AddGoalHeader
          backBtnHandler={() => {
            return;
          }}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
