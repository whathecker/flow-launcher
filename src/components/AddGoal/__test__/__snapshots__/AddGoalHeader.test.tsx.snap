// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddGoalHeader component Should renders AddGoalHeader correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "paddingLeft": 25,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingLeft": 30,
          "paddingTop": 75,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "EB-Garamond",
            "fontSize": 25,
            "fontWeight": "bold",
            "paddingRight": 12,
          },
        ]
      }
    >
      New Goal
    </Text>
    <Image
      source={1}
      style={
        Object {
          "height": 32,
          "width": 32,
        }
      }
    />
  </View>,
]
`;
